vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/treegrid/it/TreeGridScrollToPage.java

```diff
@@ -0,0 +1,63 @@
+package com.vaadin.flow.component.treegrid.it;
+
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.treegrid.TreeGrid;
+import com.vaadin.flow.data.provider.hierarchy.TreeData;
+import com.vaadin.flow.data.provider.hierarchy.TreeDataProvider;
+import com.vaadin.flow.router.Route;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Route("vaadin-grid/treegrid-scroll-to")
+public class TreeGridScrollToPage extends Div {
+
+    public TreeGridScrollToPage() {
+        TreeGrid<String> grid = new TreeGrid<>();
+        grid.setPageSize(10);
+        grid.addHierarchyColumn(String::toString).setHeader("Item");
+
+        add(grid);
+
+        TreeData<String> data = new TreeData<>();
+
+        Map<String, String> parentPathMap = new HashMap<>();
+
+        TreeGridHugeTreePage.addRootItems("Granddad", 50, data, parentPathMap)
+                .forEach(granddad -> TreeGridHugeTreePage
+                        .addItems("Dad", 3, granddad, data, parentPathMap)
+                        .forEach(dad -> TreeGridHugeTreePage.addItems("Son", 3,
+                                dad, data, parentPathMap)));
```

Comment:
The use of `TreeGridHugeTreePage` creates an implicit dependency on another test page, which may complicate future refactorings of tests.

Reply:
Addressed in a separate [PR](https://github.com/vaadin/flow-components/pull/5003).

Reply:
The refactoring PR was merged.